subsys/net/lib/sockets/Kconfig

```diff
@@ -0,0 +1,28 @@
+# Kconfig - BSD Sockets like API
+
+#
+# Copyright (c) 2017 Linaro Limited.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig NET_SOCKETS
+	bool "BSD Sockets like API"
+	default n
+	help
+	Provide BSD Sockets like API on top of native Zephyr networking API.
+
+if NET_SOCKETS
+
+config NET_SOCKETS_POSIX_NAMES
+        bool "Standard POSIX names for Sockets API"
+        default n
+        help
+        By default, Sockets API function are prefixed with "zsock_" to avoid
+        namespacing issues. If this option is enabled, they will be provided
+        with standard POSIX names like socket(), recv(), and close(), to help
+        with porting existing code. Note that close() may require a special
+        attention, as in POSIX it closes any file descriptor, while with this
+        option enaled, it will still apply only to sockets.
```

Comment:
An example is lwIP. Another example is someone finally addressing one of the big functionality holes of Zephyr - lack of native POSIX port.

> Not sure how we ever have 2 providers of bsd socket APIs, What libraries that potentially will be used with Zephyr would have implemented those APIs?

I'm not sure either, and that's good indication that best practices of namespacing should be followed, otherwise we get a situation of a crowd where everyone considers oneself exceptional and that problems of everyone else doesn't apply to them, and if they happen, it's somebody else's chore to resolve them.

> This is in most cases implemented by the OS, not libraries.

Zephyr is itself a library (libzephyr.a) which real applications (consisting of many other libraries) use to get some hardware abstraction and multitasking. Maybe in a couple of decades it'll become "the OS", but standing in a big-big crowd and pretending it's the only thing isn't the smoothest way to get there ;-).

Let's again take lwIP as an example. It defines 2 related config params:

~~~
#if LWIP_COMPAT_SOCKETS
#define accept(a,b,c)         lwip_accept(a,b,c)
#define bind(a,b,c)           lwip_bind(a,b,c)
...
#if LWIP_POSIX_SOCKETS_IO_NAMES
#define read(a,b,c)           lwip_read(a,b,c)
#define write(a,b,c)          lwip_write(a,b,c)
#define close(s)              lwip_close(s)
...
#endif
#endif
~~~

As the paragraph on which you comment describes, close() alone is enough to decide - it's irresponsible to grab that name for a code which doesn't do all that normal close() does, then it should be prefixed, then everything else should be prefixed for consistency.